### ISSUE TITLE
Login error: Failed getting Magic T value

### ISSUE THREAD

**Issue by flamablecharlie** [EionRobb/skype4pidgin#601]
Using the current build from source, and I'm getting the dreaded "Login error: Failed getting Magic T value".

Yes, I followed the flowchart. Yes, my password is correct. Yes, I can log into Skype for Web through a browser.

Here's the catch: The system I'm trying to use skype4pidgin on is "headless" and has no web browser installed. Command-line only.

So is there a solution to this?

**Comment by dequis** [comment]
Enable alt login?

>Here's the catch: The system I'm trying to use skype4pidgin on is "headless" and has no web browser installed. Command-line only.

https://www.digitalocean.com/community/tutorials/how-to-route-web-traffic-securely-without-a-vpn-using-a-socks-tunnel

**Comment by flamablecharlie** [comment]
Tried that too. No joy.

**Comment by dequis** [comment]
Alt login should give you different error messages

**Comment by flamablecharlie** [comment]
Same error message no matter what I try, unfortunately. And I can't use Firefox on that system to test with it.

**Comment by flamablecharlie** [comment]
Tried it again, new error:

skypeweb - Logging in: Authenticating
skypeweb - Login error: Login error: Authentication failed. Bad username or password. (code 40120)
skypeweb - Logging in: Signing off..

But I'm 100% certain of the username and password because they work fine from my desktop system on both Skype and skypeweb.

**Comment by KryDos** [comment]
@flamablecharlie I had the same issue previously and my username/password were working in web version but no in bitlbee. 

The issue in my case was `\` sign in the password. Escaping (`\\`) has helped, but I've changed my password to the one without `\` sign so I can avoid escaping.

**Comment by flamablecharlie** [comment]
My password, while complex, has no symbols in it. :(

**Comment by barulheira** [comment]
It just happened to me. I went to skype web site and had to confirm my security information.

All is fine again.

**Comment by flamablecharlie** [comment]
I've tried that, no joy. I think part of the problem is I'm running it on a headless server (no desktop).

**Comment by dequis** [comment]
Did you try to log in using your server's IP with socks proxy thing i linked?

**Comment by Madjosz** [comment]
I had this issue too and had to log in at https://go.skype.com/sfw and then enter my birthday. After that they wanted an e-mail address too but I aborted this and now it works again.

**Comment by neufeind** [comment]
Same here. I also had to confirm something in the webinterface and afterwards the login worked again.
Maybe the error could give a hint like "conntect failed; maybe try to login on the webinterface"?

**Comment by xss** [comment]
Looks like I am getting the same or a similar issue as @flamablecharlie, as well:

- using current version (commit c70141c) built from GIT source
- using "normal" login method, I get "Failed getting Magic T value, please try logging in via browser first"
  - tried that various times: works fine (no dialogs to confirm, no additional info asked by microsoft, the web version of skype opens just fine), and so does the login directly to the "Microsoft"/"Live" account
  - followed the flowchart to no avail
- using alternative login method, I get another error: "Login error: Authentication failed. Bad username or password. (code 40120)" -- same as @flamablecharlie 
- username and password were definitely correctly entered, and neither has symbols or anything extraordinary

My skype account has only one alias, and "old" skype name back from a time when they didn't have to be email addresses. This alias had worked with skypeweb before for many months, but stopped working now.

What is likely related to my issue: 
A few days ago, I received an email from Microsoft informing me that, allegedly because of new regulations for EU residents, I would need to add my birthday to my account in order to be able to use Skype. As I had done that a few months back already, I wasn't asked for that upon login to web.skype.com. I was, however, forced to create a new "alias", and used my associated email address as the alias, and Microsoft forced me (by withholding any option to do otherwise) to set this as my "primary alias", against my will.

Ever since creating that second alias, I started getting the "Magic T value" error upon login, both with old and with new alias. Using the new alias (the email address), I was at least able to log into skypeweb through the alternative login method. Both aliases were set to allowed to be used for logging into the Microsoft account at that time.

As I don't want nor need more than one alias, and actually prefer to keep my original alias as this is the one that was communicated to partners, I managed with some fiddling to set the old alias back as primary, and remove the second, new alias. However, the issue remains: I cannot login to skypeweb with the original (and now again only) alias, anymore, with neither login method, see above.

When using web log in, neither web.skype.com nor Microsoft/Live account ask for any additional info. I can log in fine and use and manage everything as expected. However, not so with skypeweb/skype4pidgin. :(

Are there possibly any logs or so I could provide to help debugging? I really don't want to have to run the Skype messenger additionally all the time.

**Comment by szr8** [comment]
I am seeing the same as @flamablecharlie and am also using c70141ca8a6d146bc759aa293a3f79132bd248c7 where many of my messages (seemingly randomly) just don't go through.

**Comment by brianjmurrell** [comment]
I'm now getting this on an account/configuration that has been working for many many months.  Just all of a sudden within the last 30-60 minutes I got disconnected and any attempt to connect yields this error.

Using https://web.skype.com/ works fine.  I opened that up and was not asked for anything and was just taken to my conversations.

**Comment by brianjmurrell** [comment]
Any ideas here?  I've been getting this for 4 days now.  Should I open a new ticket?  Perhaps this one is stale and being ignored now?

**Comment by brianjmurrell** [comment]
Hrm.  So, had to actually use the incognito window login trick from the flowchart.  Using a regular (non-incognito) window didn't resolve it.

**Comment by dbeusee** [comment]
checking "use alternative login method" solved this error for me.  Then I noticed when I use firefox to log in, it prompts for the "no password" option MS is pushing now.  I clicked no thanks and unchecked "use alternative login method" and it is working again.  I also noticed this message:
Support for this version of Skype for Web is coming to an end. A new preview version is available now for Microsoft Edge and Google Chrome with HD video calling, call recording and much more. Try it out

I hope this upcoming change doesn't affect skypeweb, but it probably will.

**Comment by coozoo** [comment]
Strange got this error week ago, usually  browser login was helpfull but not now... and I can't find such option like "alternative login method"...

**Comment by dbeusee** [comment]
You need to log in with a browser you haven't used on the skype web site or use an incognito window (chrome's menu has "new incognito window").  I think IE and Edge browsers have a similar thing called InPrivate browsing.  Firefox has "New Private Window".  Use one of these to log into skype web and click the "no thanks" link for the no password prompt.  Or in pidgin, you can go to accounts -> manage account, then double-click the skype account, then advanced tab, and check "use alternative login method".

**Comment by AleXoundOS** [comment]
Alternative login method worked for me as of now. Will see how reliable it is.

**Comment by coozoo** [comment]
Thanks for steps.. but damn I've did  nothing and today it was simply reconnected succesfuly... Microsoft what they doing there... hope it won't happen to github..

**Comment by prosoitos** [comment]
Thank you @KryDos: I was also having a problem with bitlbee and it turned out that I had the same issue as you did, except that it was not with `\`, but with `'` and `"`. So those 2 special characters are also problematic. Removing them from my password solved it for me.

**Comment by vrobolab** [comment]
I'm getting that error right now. Password does not contain any special characters - just a basic latin and digits.

**Comment by vrobolab** [comment]
@vrobolab logging in to web.skype.com in a new (clean) browser profile solved the issue - shortly after I logged in into the browser the "magic t value" error disappeared and pidgin connected too.

**Comment by jticket1024** [comment]
Logging in to [](https://login.live.com/login.srf?wa=wsignin1.0&rpsnv=13&ct=1564576517&rver=6.7.6655.0&wp=SA_20MIN&wreply=https%3A%2F%2Faccount.live.com%2FActivity%3Fuaid%3D3a3c4afb2b5348eebfa73092758c1598&lc=1033&id=38936&mkt=en-US&uaid=3a3c4afb2b5348eebfa73092758c1598) and confirming the sign in usually fixes the magic T value problem for me.

**Comment by brianjmurrell** [comment]
I had an instance where I was being refused login due to _perceived_ location changing drastically (due to the use of a VPN on another host with the same account being used not in the VPN) and had to _verify_ my MS account.

**Comment by jiteshsharma** [comment]
Getting the same issue this morning. Failed to get magic T-Value. Logged in successfully from Chrome incognito/private window. Still no luck.

alt-login not enabled as 2FA not enabled on Skype.

**Comment by brianjmurrell** [comment]
This seems to be back today.  Logged in via skypeweb and got a message saying ToS were changing.  I clicked _Next_ and was taken to the main UI -- good to go.  Logging in via skype4pidgin still getting this error.

Using alt-login method triggers #643.

**Comment by brianjmurrell** [comment]
And now it's working again.  ~sigh~

**Comment by vrobolab** [comment]
Not working for me :( 
"Failed getting Magic T value" again, but there is no ToS screen if I login with the browser.

**Comment by alexolog** [comment]
Not working for me.  I think that maybe some of the servers were not yet updated.

**Comment by coozoo** [comment]
the same sh*t happened for me from yesterday and now login relogin doesn't help. This dam* microsoft dorks with their sh*tty skype 3GB in RAM just hard imagine but nope here it is microsoft...

**Comment by kbalicki** [comment]
"Failed getting Magic T value, please try logging in via browser first" again today, old solutions do not work :/

**Comment by brianjmurrell** [comment]
As quickly as it was working again yesterday, again today, it's broken.

**Comment by SasakiSadako** [comment]
Since yesterday i have the same issue. Seems to be not possible to use skype in pidgin for now. I hope it could be solved again :)

**Comment by SasakiSadako** [comment]
im also tried to login with the alternate login option.
I just get the following error:
Login error: No handler found for resource (code 40499)

I also tried to login on the browser, and also logged in with my regular skype on the Device. "Thats usually helped me before with the magic T error.
I updated the plugin to 1.6 but still cannot login. Im running it on KDE Neon with the newest focal upgrade.
So still no skype for me on pidgin
